src/components/Banners/SlackNotificationBanner.tsx

```diff
@@ -0,0 +1,215 @@
+    UpdateUserSettingsMutation,
+    UpdateUserSettingsMutationVariables
+  >(UPDATE_USER_SETTINGS, {
+    onCompleted: () => {
+      hideBanner();
+      dispatchBanner.successBanner(
+        "You will now receive Slack notifications when your patches fail or succeed"
+      );
+    },
+    onError: (err) => {
+      dispatchBanner.errorBanner(
+        `Error while saving settings: '${err.message}'`
+      );
+    },
+    refetchQueries: ["GetUserSettings"],
+  });
+
+  const saveNotificationSettings = () => {
+    if (!slackUsername) {
+      return;
+    }
+    updateUserSettings({
+      variables: {
+        userSettings: {
+          slackUsername,
+          notifications: {
+            patchFinish: "slack",
+            patchFirstFailure: "slack",
+          },
+        },
+      },
+    });
+  };
+
+  const cookieExists = cookie === doNotShowslackNotificationBannerCookie;
+
+  const showSlackBanner =
+    !cookieExists &&
+    settingsSlackNotificationPatchFail === "" &&
+    settingsSlackNotificationPatchFinish === "";
+
+  return (
+    <StyledPageWrapper>
+      <Banners
+        banners={bannersState}
+        removeBanner={dispatchBanner.removeBanner}
+      />
+      {showSlackBanner && (
+        <StyledBanner
+          bannerTheme="information"
+          data-cy="slack-notification-banner"
+        >
+          <Wrapper>
+            <StyledIcon glyph="InfoWithCircle" fill={uiColors.blue.base} />
+            <StyledBody>
+              You can receive a Slack notification when your patch is ready.
+            </StyledBody>
+            <Popconfirm
+              title={
+                <>
+                  <SlackUsername weight="medium">Slack Username</SlackUsername>
+                  <Input
+                    data-cy="slack-username-input"
+                    value={slackUsername}
+                    onChange={(e) => setSlackUsername(e.target.value)}
+                    onPressEnter={() => saveNotificationSettings()}
+                  />
+                </>
+              }
+              onConfirm={() => saveNotificationSettings()}
+              onCancel={() => setSlackUsername("")}
+              okText="Save"
+              cancelText="Cancel"
+              okButtonProps={{ loading: loadingUpdateUserSettings }}
+              cancelButtonProps={{ disabled: loadingUpdateUserSettings }}
+              icon={null}
+            >
+              <div data-cy="subscribe-to-notifications">
+                <SubscribeButton>Subscribe</SubscribeButton>
+              </div>
+            </Popconfirm>
+          </Wrapper>
+          <IconButton
+            data-cy="dismiss-slack-notification-banner"
+            aria-label="Close Slack Notfication Banner"
+            onClick={hideBanner}
+          >
+            <Icon glyph="X" fill={uiColors.blue.base} />{" "}
+          </IconButton>
+        </StyledBanner>
+      )}
+    </StyledPageWrapper>
+  );
+};
+
+const StyledBanner = styled(Banner)`
```

Comment:
I really like how this banner looks we should adopt a similar style for the other banners!